test/CancellationHandleTest.cpp

```diff
@@ -106,3 +116,170 @@ TEST(CancellationHandle, verifyCancellationWorksWithMultipleThreads) {
       CancellationException);
   EXPECT_TRUE(handle.isCancelled());
 }
+
+// _____________________________________________________________________________
+
+TEST(CancellationHandle, ensureObjectLifetimeIsValidWithoutWatchDogStarted) {
+  EXPECT_NO_THROW(CancellationHandle<true>{});
+}
```

Comment:
Actually, now that I am looking at this:
Why don't you ALWAYS start the watchdog in the constructor? That two phase initialization always makes the call site more cluttered.